src/vsphere_cpi/lib/cloud/vsphere/drs_rules/vm_group.rb

```diff
@@ -59,15 +59,53 @@ def delete_vm_groups(vm_group_names)
           group_spec.remove_key = vm_group.name
           group_spec
         end
+
+        # If these vm_groups are getting deleted. Also delete all the
+        # vm-host group rules associated with it.
+        #
+        # @TODO : CAVEAT: if anyone creates a rule outside of CPI using these
+        # vm_groupS, IT will be deleted too. To narrow
+        # the scope of this caveat, CPI will only delete the
+        # vm-host-group-affinity rule type. This is because, currently it only
+        # creates vm-host-affinity rule using vm groups.
+        #
+        # Long Term Solution is to disallow admin to specify their own rule
+        # names and CPI generates a rule name from vm_group and hence will be
+        # able to identify correct rule to be deleted.
+        rules_spec = get_delete_rule_spec(empty_vm_groups) unless empty_vm_groups.nil?
+        logger.debug("Deleting #{rules_spec.count} vm host affinity rules") if rules_spec
         config_spec = VimSdk::Vim::Cluster::ConfigSpecEx.new
         config_spec.group_spec = group_specs
+        config_spec.rules_spec = rules_spec
         @client.wait_for_task do
           @cluster.reconfigure_ex(config_spec, true)
         end
       end
     end
 
     private
+    def get_delete_rule_spec(empty_vm_groups)
+      if @cluster.configuration_ex.rule.nil?
+        return nil
+      end
+      rules_spec= empty_vm_groups&.each_with_object([]) do |vm_group, rules_spec|
```

Comment:
dont need an & check as empty_vm_groups is an array and the check is being made when this is set